docs/excel/performance.md

```diff
@@ -104,8 +104,145 @@ Excel.run(async (ctx) => {
+
+### Payload size
+
+The payload size of a request is a combination of the following three components.
+
+* The number of API calls.
+* The number of objects, such as `Range` objects.
+* The length of the value to set or get.
+
+If an API returns the `RequestPayloadSizeLimitExceeded` error, use the best practice strategies documented in this article to optimize your script and avoid the error.
+
+### Strategy 1: Move unchanged values out of loops
+
+Limit the number of processes that occur within loops to improve performance. In the following code sample, `context.workbook.worksheets.getActiveWorksheet()` can be moved out of the `for` loop, because it doesn't change within that loop.
+
+```js
+// DO NOT USE THIS CODE SAMPLE. This sample shows a poor performance strategy. 
+async function run() {
+  await Excel.run(async (context) => {
+    var ranges = [];
+    
+    // This sample retrieves the worksheet every time the loop runs, which is bad for performance.
+    for (let i = 0; i < 7500; i++) {
+      var rangeByIndex = context.workbook.worksheets.getActiveWorksheet().getRangeByIndexes(i, 1, 1, 1);
+    }    
+    await context.sync();
+  });
+}
+```
+
+The following code sample shows logic similar to the preceding code sample, but with an improved performance strategy. The value `context.workbook.worksheets.getActiveWorksheet()` is retrieved before the `for` loop, because this value doesn't need to be retrieved each time the `for` loop runs. Only values that change within the context of a loop should be retrieved within that loop.
+
+```js
+// This code sample shows a good performance strategy.
+async function run() {
+  await Excel.run(async (context) => {
+    var ranges = [];
+    // Retrieve the worksheet outside the loop.
+    var worksheet = context.workbook.worksheets.getActiveWorksheet(); 
+
+    // Only process the necessary values inside the loop.
+    for (let i = 0; i < 7500; i++) {
+      var rangeByIndex = worksheet.getRangeByIndexes(i, 1, 1, 1);
+    }    
+    await context.sync();
+  });
+}
+```
+
+### Strategy 2: Create fewer range objects
+
+Create fewer range objects to improve performance and minimize payload size. Two approaches for creating fewer range objects are described in the following article sections and code samples.
+
+#### Split each range array into multiple arrays
+
+One way to create fewer range objects is to split each range array into multiple arrays, and then process each new array with a loop and a new `context.sync()` call.
+
+> [!IMPORTANT]
+> Only use this strategy you've first determined that you're exceeding the payload request size limit. Using multiple loops can reduce the size of each payload request to avoid exceeding the 5MB limit, but using multiple loops and multiple `context.sync()` calls also negatively impacts performance.
```

Comment:
```suggestion
> Only use this strategy if you've first determined that you're exceeding the payload request size limit. Using multiple loops can reduce the size of each payload request to avoid exceeding the 5MB limit, but using multiple loops and multiple `context.sync()` calls also negatively impacts performance.
```